src/Controls/src/Core/TrimmerConstants.cs

```diff
@@ -0,0 +1,9 @@
+namespace Microsoft.Maui.Controls;
+
+class TrimmerConstants
+{
+    // https://github.com/dotnet/runtime/blob/f130138b337b57342e94dabf499b818531effed5/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs#L31-L32
+    internal const string SerializerTrimmerWarning = "Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the required types are preserved.";
+
+    internal const string NativeBindingService = "This method properly handles missing properties, and there is not a way to preserve them from this method.";
```

Comment:
It's good that the code handles missing properties, the question is if missing property can cause application behavior change? If yes - then this is still a problem.

For example:
* Serializers can handle missing properties just fine, they just don't write them out. But this can break apps as the deserialization of such payload may fail due to the missing property.
* On the other hand, a debugger visualizer can handle missing properties and it will not break the app if it doesn't show a value for unused property.

Reply:
Yeah this one might cause an issue...

I think this is a feature that came from Xamarin.Forms: 

https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-views/xaml

So you might have `.xaml` like:
```
<ios:UILabel Text="{Binding MyProperty}" />
```

But then, I don't think that `UILabel.Text` would actually get linked away because it's used by other Maui assemblies (that aren't trimmed). This type is in `Xamarin.iOS.dll` with all the native UIKit APIs, which *is* trimmable.

So seems like it would only break if:

1. You tried to binding against a specific property Maui doesn't use at all.
2. You have XamlC off.

So maybe users aren't likely to hit this -- and this feature is not probably used much. I would probably be fine to see if anyone runs into this during future previews.

Reply:
Sounds reasonable - with my comment I just wanted to point out that not crashing is kind of the "min bar" :wink: